src/api/CompleteChecklistItem.ts

```diff
@@ -0,0 +1,63 @@
+import { ICheckListItem } from "api/CheckListItemApi";
+import { spWebContext } from "providers/SPWebContext";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { DateTime } from "luxon";
+import { Person, useCurrentUser } from "api/UserApi";
+
+const completeCheckListItem = (
+  item: ICheckListItem,
+  checklistItems: ICheckListItem[] | undefined,
+  currentUser: Person
+) => {
+  const [batchedSP, execute] = spWebContext.batched();
+  const batch = batchedSP.web.lists.getByTitle("CheckListItems");
+
+  // Always add the current update to the batch
+  batch.items.getById(item.Id).update({
+    CompletedById: currentUser.Id,
+    CompletedDate: DateTime.now().toISODate(),
+  });
+
+  // Find additional updates
+  switch (item.TemplateId) {
+    // Valid for testing only
+    // Welcome Package 1 should enable TESTING ITEM -1
+    case 1: //Testing only
+      checklistItems?.forEach((element) => {
+        if (element.TemplateId === -1) {
+          batch.items.getById(element.Id).update({ Active: true });
+        }
+      });
+      break;
+
+    default:
+      break;
+  }
+  return execute();
+};
+
+export const useCompleteChecklistItem = (item: ICheckListItem) => {
+  const queryClient = useQueryClient();
+  const checklistItems = queryClient.getQueryData<ICheckListItem[]>([
+    "checklist",
+    item.RequestId,
+  ]);
+  const currentUser = useCurrentUser();
+
+  return useMutation(
+    ["checklist", item.RequestId],
+    () => {
+      if (process.env.NODE_ENV === "development") {
+        // TODO: Find a better way to show this in nodejs dev environment
+        return Promise.resolve();
+      } else {
+        return completeCheckListItem(item, checklistItems, currentUser);
+      }
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(["checklist"]);
+      },
```

Comment:
Add some onErrror logic, so we don't have to come back and rework this later?

Reply:
Also, along this thought...  Other places I've used onSettled, this fires after onSuccess and onError, to refresh the query and re-render so they can attempt action again if the error was something they can try again, or to show them the "new" data if the error was a conflict.  However, I supposed we could add text to errors to say to refresh the page and try again, and if the error persists, please input a help request.